concourse/pipelines/create-bosh-cloudfoundry.yml

```diff
@@ -0,0 +1,901 @@
+                  path: existing-bosh-CA
+              outputs:
+                - name: generated-bosh-CA
+              run:
+                path: sh
+                args:
+                  - -e
+                  - -c
+                  - |
+                    if  [ -z "$(tar -tvzf existing-bosh-CA/bosh-CA.tar.gz)" ] ; then
+                      certstrap init --passphrase "" --common-name bosh-CA
+                      cd out
+                      tar -cvzf ../generated-bosh-CA/bosh-CA.tar.gz bosh-CA.*
+                      cd ..
+                    else
+                      echo "The CA cert already exists, skipping generation..."
+                      cp existing-bosh-CA/bosh-CA.tar.gz generated-bosh-CA/bosh-CA.tar.gz
+                      mkdir out
+                      tar -xvzf generated-bosh-CA/bosh-CA.tar.gz -C out
+                    fi
+                    ./paas-cf/concourse/scripts/file-to-yaml.sh secrets bosh_ca_cert out/bosh-CA.crt > generated-bosh-CA/bosh-ca-cert.yml
+            on_success:
+              put: bosh-CA
+              params:
+                file: generated-bosh-CA/bosh-CA.tar.gz
+          - task: extract_terraform_outputs
+            config:
+              platform: linux
+              image: docker:///ruby#2.2-slim
+              inputs:
+                - name: paas-cf
+                - name: vpc-tfstate
+                - name: bosh-tfstate
+              outputs:
+                - name: terraform-outputs
+              run:
+                path: sh
+                args:
+                  - -c
+                  - -e
+                  - |
+                    ruby paas-cf/concourse/scripts/extract_terraform_state_to_yaml.rb \
+                      < vpc-tfstate/vpc.tfstate \
+                      > terraform-outputs/vpc.terraform-outputs.yml
+                    ruby paas-cf/concourse/scripts/extract_terraform_state_to_yaml.rb \
+                      < bosh-tfstate/bosh.tfstate \
+                      > terraform-outputs/bosh.terraform-outputs.yml
+
+      - task: render-manifest
+        config:
+          platform: linux
+          image: docker:///governmentpaas/spruce
+          inputs:
+            - name: paas-cf
+            - name: terraform-outputs
+            - name: generated-bosh-secrets
+            - name: generated-bosh-CA
+          outputs:
+            - name: bosh-manifest
+          params:
+            BOSH_MANIFEST_STUBS: |
+              ./generated-bosh-secrets/bosh-secrets.yml
+              ./generated-bosh-CA/bosh-ca-cert.yml
+              ./terraform-outputs/bosh.terraform-outputs.yml
+              ./terraform-outputs/vpc.terraform-outputs.yml
+          run:
+            path: sh
+            args:
+              - -e
+              - -c
+              - |
+                ./paas-cf/manifests/bosh-manifest/build_manifest.sh $BOSH_MANIFEST_STUBS > bosh-manifest/bosh-manifest.yml
+                /bin/cat bosh-manifest/bosh-manifest.yml
```

Comment:
That's a valid question, but this was present in the existing pipeline, so I left it.